repose-aggregator/core/repose-core-api/src/main/java/org/openrepose/core/services/RequestProxyService.java

```diff
@@ -31,17 +31,38 @@
     //TODO: this is the most terrible return value ever
     int proxyRequest(String targetHost, HttpServletRequest request, HttpServletResponse response) throws IOException;
 
+    int proxyRequest(String targetHost, HttpServletRequest request, HttpServletResponse response, String connPoolId)
+            throws IOException;
+
     void setRewriteHostHeader(boolean value);
 
+    @Deprecated
```

Comment:
Yes, the DD is currently the only Repose component which uses this API, however it is exposed as a "service" so there is the potential that a user has developed something which also uses this API.